uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/UaaOrchestratorClaimTest.java

```diff
@@ -0,0 +1,301 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.security.oauth2.client.DefaultOAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.token.DefaultAccessTokenRequest;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = OrchestratorIntegrationTestConfig.class)
+@Slf4j
+public class UaaOrchestratorClaimTest {
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOGGER = LoggerFactory.getLogger(UaaOrchestratorClaimTest.class);
+
+    public static final String ZONES_URI = "/orchestrator/zones";
+
+    private UaaRequest uaaRequest;
+
+    protected static HttpHeaders headers;
+    Random random = new Random();
+
+    private  final String UAA_SUBDOMAIN = "rest-uaa-subdomain"+random.nextInt(1000);
+
+    private static final String UAA_ADMIN_CLIENT_SECRET = "ZHVtbXlBZG1pblNlY3JldDEyMyMh";
+
+    private static final String UAA_CLAIM_NAME_PREFIX = "uaa-it-";
+    @Value("${uaa.url}")
+    private String uaaUrl;
+
+    @Value("${uaa.orchestrator.clientId}")
+    private String clientId;
+
+    @Value("${uaa.orchestrator.clientSecret}")
+    private String clientSecret;
+
+    @Value("${kubernetes.config.env}")
+    private String environment;
+
+    @Autowired
+    private OrchestratorUtil orchestratorUtil;
+
+    protected static OAuth2RestTemplate trustedIssuerValidScopesRestTemplate;
+
+    @Value("${kubernetes.config.secondary-namespace:iam-dev}")
+    private String namespace;
+
+
+    private Integer uaaInstanceCount = random.nextInt(1000);
+    ;
+
+
+    // Below regex UMS_EKS_ENV_URI_REG will match all UMS EKS Env uris. Examples, https://user-management-int.pss.dev.usw02.15.energy
+    // https://user-management-dev.pss.dev.usw02.15.energy, https://user-management-stg.pss.dev.usw02.15.energy
+    private static String UMS_EKS_ENV_URI_REG =
+            "(https://[-a-zA-Z0-9]*(\\.uaa-qa\\.pss\\.dev\\.usw02\\.15\\.energy){1,1})";
```

Comment:
This regex wont work for int and stage envs if tests executed against int and stage, can you change it to cover all three envs int, qa and stage.

Reply:
Done